### DISCUSSION TITLE
Camera jitters while players moves

### DISCUSSION THREAD

**Discussion by HelloDrak** [ramokz/phantom-camera#231] [Question / Help Wanted] [answered]
I just set up a basic scene with 2DCam and whenever I move in *any* follow mode...the camera jitters pretty bad. This isnt happening with the regular Godot camera and only seems to happen when I fully setup PhantomCamera (PC2D, PCCameraHost + Camera2D)

I tried this with all follow modes, so Im pretty stumped on what might be causing this.

A few test results:
1. Pixel Perfect seems to make the jittering worse
2. I installed the smoothing addon...which also makes the issue worse.
3. Im using the current stable version and tested this in 4.2.1 and 4.3Dev4, with the same issues.

Ive attached images and a video that should hopefully help with understanding what's going on. Any help is greatly appreciated!!!

![Capture](https://github.com/ramokz/phantom-camera/assets/6536265/e7d242e6-a92f-4cf9-8c21-fbdc638bdab7)

https://github.com/ramokz/phantom-camera/assets/6536265/eaf4fe30-9c24-4dfa-bc87-e40cf7eee2f2

**Comment by ramokz** [comment]
How does the movement logic work? Are you moving the character in the `_process` or `_physics_process` method?

**Reply by HelloDrak** [reply]
All the movement logic is in `_process`. (The update_animation_parameters() is just toggling the animation tree state)

![image](https://github.com/ramokz/phantom-camera/assets/6536265/0c4cafaa-3102-4ee1-9213-adab31bcd516)

**Reply by ramokz** [reply, marked as answer]
Ah that's likely the cause of it then!

Generally speaking, character movement should be occurring in the `_physics_process`. So if you run that code using that it _should_ be working correctly.

**Reply by HelloDrak** [reply]
You are amazing - that did it! Thank you for solving my issue...which seems to have nothing to do with PhantomCamera lol

And thank you for making such an awesome addon!!

https://github.com/ramokz/phantom-camera/assets/6536265/2afab9ba-44c0-42d0-804c-bf5c1ea938a9

**Reply by ramokz** [reply]
No problem! Glad it was a simple solve 😄
